types/astring/test/astring-test.ts

```diff
@@ -0,0 +1,38 @@
+import { FunctionExpression, MemberExpression, Program } from "estree";
+import { Stream } from "stream";
+import { baseGenerator, generate } from "..";
+
+const ast: Program = null;
+const functionE: FunctionExpression = null;
+const memberE: MemberExpression = null;
+
+//should accept different nodes
+generate(ast);
+generate(functionE);
+generate(memberE);
+
+//global scope function
+astring.generate(ast);
+
+//options without output option should generate string
+const string: string = generate(ast, {
+    comments: true,
+    generator: baseGenerator,
+    indent: "\t",
+    lineEnd: "\n",
+    startingIndentLevel: 42,
+    sourceMap: null
+});
+
+//options with output option should return Stream
+const stream: Stream = generate(ast, {
+    output: new Stream()
+});
+
+//Generator should map node types to functions whose first parameter is same node type
+baseGenerator.Program(ast, { write: function (s: string) { return; } });
+baseGenerator.FunctionExpression(functionE, { write: function (s: string) { return; } });
+baseGenerator.MemberExpression(memberE, { write: function (s: string) { return; } });
+
+//global scope function
+astring.baseGenerator.Program(ast, { write: function (s: string) { return; } });
```

Comment:
Move global test to `YourLibraryName-global.test.ts` as per faq https://github.com/DefinitelyTyped/DefinitelyTyped#how-do-i-write-definitions-for-packages-that-can-be-used-globally-and-as-a-module